### ISSUE TITLE
Angular *ngFor breaks embla ("ascDiffsToSnaps[0] is undefined")

### ISSUE THREAD

**Issue by nnvnzndrvn** [davidcetinkaya/embla-carousel#71]
I managed to get Embla to work with Angular the following way but I had the following issue:

Angular version: 
`Angular: 9.1.11`

Import it in the component (no need to import in AppModule apparently):
`import EmblaCarousel from 'embla-carousel';`

Define a ViewChild to get the elementNode:
 ```
 @ViewChild('emblabehaviors', {static: true}) protected emblaNode: ElementRef<any>;
 embla: EmblaCarousel;
```

And initialize in ngOnInit():
```
 this.embla = EmblaCarousel(this.emblaNode.nativeElement, {
      draggable: true,
      dragFree: true,
      loop: false,
      speed: 1
 });
```

The html:
 - this one works
 
 ```
 <div #embla class="embla">
    <div class="embla__container">
            <div class="embla__slide slider-element">Thing 1</div>
            <div class="embla__slide slider-element">Thing 2</div>
            <div class="embla__slide slider-element">Thing 3</div>
     </div>
</div>
```

 - this one returns "ERROR TypeError: "ascDiffsToSnaps[0] is undefined"" when starting to drag:
 ```
 <div #embla class="embla">
    <div class="embla__container">
            <div class="embla__slide slider-element" *ngFor="let thing of things">
                {{thing.name}}
            </div>
     </div>
</div>
```

Full error:
```
ERROR TypeError: "ascDiffsToSnaps[0] is undefined"
    findTargetSnap index.js:1861
    byDistance index.js:1898
    seekTargetBy index.js:1236
    up index.js:1297
    Angular 7
        invokeTask
        onInvokeTask
        invokeTask
        runTask
        invokeTask
        invokeTask
        globalZoneAwareCallback
core.js:6228:19
    Angular 4
        defaultErrorLogger
        handleError
        next
        schedulerFn
    RxJS 5
        __tryOrUnsub
        next
        _next
        next
        next
    Angular 11
        emit
        onHandleError
        invoke
        run
        runOutsideAngular
        onHandleError
        handleError
        runTask
        invokeTask
        invokeTask
        globalZoneAwareCallback
```

It seems this issue only happens when using *ngFor to iterate over an array. If you point me in the right direction I can try to investigate further but I'm not a pro. Great carousel btw!

**Comment by davidcetinkaya** [comment]
Hello @nnvnzndrvn,

Thank you for picking this up 👍. That's strange. What version of Embla are you using? Shouldn't the `ngFor` loop be on the container in order to loop all slides like below?

```javascript
<div #embla class="embla">
  <div class="embla__container" *ngFor="let thing of things">
    <div class="embla__slide slider-element">
      {{thing.name}}
    </div>
  </div>
</div>
```

It would be really helpful if you could setup a CodeSandbox with Angular so I can debug it further 🙂.

Best,
David

**Comment by nnvnzndrvn** [comment]
I'm using
`  "_id": "embla-carousel@3.0.2"`

I think the `*ngFor` should be on the `embla__slider `class, otherwise it will create many containers.

Generally the library can be used fine, but issues happen when using `*ngIf` conditions. If I remember correctly, these take the element out of the DOM when the condition is false, and probably embla is unable to find it then. This can be worked around by using` [ngClass]` with a .css class that renders the slide invisible and unusable, for example:

```
.hidden {
  pointer-events: none;
  opacity: 0;
  z-index: -1000;
  width: 0px;
  height: 0px;
}
```

But being able to use `*ngIf` conditions would be much more convenient. I created this sandbox, but for some reason it's returning an error message when initializing the elma carousel:

`ERROR TypeError: embla_carousel_1.default is not a function`

On my local environment all's good, I'm not very familiar with CodeSandbox, so maybe I'm overlooking something. I'll create a repo on github during the weekend that you can npm install and ng serve to check it.

**Comment by davidcetinkaya** [comment]
Hello again @nnvnzndrvn,

Here's a [working example](https://codesandbox.io/s/embla-test-angular-kekc2). The issue was not related to Embla, it was due to the invalid import statement. Seems like there is no [esModuleInterop](https://stackoverflow.com/questions/56238356/understanding-esmoduleinterop-in-tsconfig-file) flag in your sandbox, so changing this line:

```typescript
import EmblaCarousel from 'embla-carousel';
```

...into this:

```typescript
import * as EmblaCarousel from "embla-carousel";
```

Made it work. Let me know if this helps!

Kindly
David

**Comment by nnvnzndrvn** [comment]
Hi @davidcetinkaya,

Thanks! I see, I wasn't aware of this esModuleInterop :)

I modified it to include one example using `*ngIf` and another one with `[ngClass]`. [Here](https://codesandbox.io/s/embla-test-angular-3ise5?file=/src/carousel/carousel.component.html).

I tried on ngAfterViewInit but it's the same result. I guess that the `*ngIf` is read before Elma can catch it. I followed what's suggested [here](https://stackoverflow.com/questions/39366981/viewchild-in-ngif), and I managed to get it to work. However, when switching, at some point one of them gets bugged and stops.

**Comment by davidcetinkaya** [comment]
Hello @nnvnzndrvn,

I've released a patch fix with [version 3.0.23](https://github.com/davidcetinkaya/embla-carousel/releases/tag/v3.0.23). Dragging a carousel with no slides shouldn't trigger the error you mention anymore. Just wanted to let you know if you want to experiment with Embla and Angular more.

Kindly,
David

**Comment by davidcetinkaya** [comment]
I'm closing this for now due to lack of activity. Feel free to re-open this issue if you want to pursue this.

Best,
David
